Tools/CISupport/ews-build/steps.py

```diff
@@ -86,7 +86,95 @@ def pr_url(cls, pr_number, repository_url=None):
             return ''
         if not pr_number or not isinstance(pr_number, int):
             return ''
-        return '{}pull/{}'.format(repository_url, pr_number)
+        return '{}/pull/{}'.format(repository_url, pr_number)
+
+    @classmethod
+    def api_url(cls, repository_url=None):
+        if not repository_url:
+            repository_url = '{}{}'.format(GITHUB_URL, GITHUB_PROJECTS[0])
+
+        if repository_url not in GitHub.repository_urls():
+            return ''
+        _, url_base = repository_url.split('://', 1)
+        host, path = url_base.split('/', 1)
+        return 'https://api.{}/repos/{}'.format(host, path)
+
+    @classmethod
+    def credentials(cls):
+        try:
+            passwords = json.load(open('passwords.json'))
+            return passwords.get('GITHUB_COM_USERNAME', None), passwords.get('GITHUB_COM_ACCESS_TOKEN', None)
+        except Exception as e:
+            print('Error reading GitHub credentials')
+            return None, None
+
+
+class GitHubMixin(object):
+    addURLs = False
+    pr_open_states = ['open']
+    pr_closed_states = ['closed']
+
+    def fetch_data_from_url_with_authentication(self, url):
+        response = None
+        try:
+            username, access_token = self.credentials()
+            auth = HTTPBasicAuth(username, access_token) if username and access_token else None
+            response = requests.get(
+                url, timeout=60, auth=auth,
+                headers=dict(Accept='application/vnd.github.v3+json'),
+            )
+            if response.status_code // 100 != 2:
+                self._addToLog('stdio', 'Accessed {url} with unexpected status code {status_code}.\n'.format(url=url, status_code=response.status_code))
+                return None
+        except Exception as e:
+            # Catching all exceptions here to safeguard access token.
+            self._addToLog('stdio', 'Failed to access {url}.\n'.format(url=url))
+            return None
+        return response
+
+    def get_pr_json(self, pull_request, repository_url=None):
```

Comment:
pull_request => pr_number (please check for this at other places as well)
(pull_request gives an impression that it is an object instead of an integer)